File: boj/DynamicProgramming/BOJ_2670.py
# 백준 2670 연속부분최대곱
# 21.02.22

import sys
from functools import cmp_to_key
from collections import deque
from itertools import combinations
from itertools import permutations
sys.setrecursionlimit(10**6)

n = int(input())
arr = []
for _ in range(n):
    arr.append(float(sys.stdin.readline().rstrip()))


for i in range(1, n):
    arr[i] = max(arr[i - 1] * arr[i], arr[i])

print("%.3f" %max(arr))

File: boj/DynamicProgramming/BOJ_2670.swift
// 백준 2670 연속부분최대곱
// 21.02.23

import Foundation

let n = Int(readLine()!)!
var a: [Double] = []

for _ in 0..<n {
  a.append(Double(readLine()!)!)
}

for i in 1..<n {
  if a[i] < a[i] * a[i - 1] { a[i] *= a[i - 1] }
}

let ans = String(format: "%.3f", (a.max()! * 1000).rounded() / 1000)
print(ans)
